### ISSUE TITLE
Wrong start system reported to EDDN with scanned body data.

### ISSUE THREAD

**Issue by Tkael** [EDCD/EDDI#941]
Per Discord report: 
```
EDDI is still occasionally sending Scan events where the system doesn't match the body.  See https://docs.google.com/spreadsheets/d/1BNn00XjyeRreLJrN2Tc_JOTEqM3enkVqKplD84BujSA/edit#gid=2110804481
```

**Comment by Tkael** [comment]
Theory: An exception is thrown during a location generating event. Updated location is not passed correctly to the EDDN listener, and it consequently reports the commander's prior star system as the current star system.

**Comment by klightspeed** [comment]
Example events from the last 24 hours:
https://ross.eddb.io/eddn/log/112059519

https://ross.eddb.io/eddn/log/111889347
https://ross.eddb.io/eddn/log/111890396
https://ross.eddb.io/eddn/log/111891309
https://ross.eddb.io/eddn/log/111895568
https://ross.eddb.io/eddn/log/111895704
https://ross.eddb.io/eddn/log/111896434 

It does look like FSDJumps are sometimes being missed.

The FSDJump for the first example never made it to EDDN.

In the second group of examples, EDDiscovery was running in parallel.  Refer https://docs.google.com/spreadsheets/d/1YVEohPXOml-0LULN3sapPI4pJveGzPoC4JroFLQKlag/edit#gid=0
Referring to the EDDN source code, the gateway should not return any error when a duplicate message is encountered - it appears to be the relay that discards duplicate messages.

It is possible that either the file size is being reported inaccurately (e.g. rounded to the nearest 4kB), or the game sometimes does not atomically write an entry and so EDDI sometimes reads a partial entry.
We encountered the same problems in 2016 with netlogs in EDDiscovery, which is why we switched to a custom log file reader - see https://github.com/EDDiscovery/EDDiscovery/blob/master/EliteDangerous/EliteDangerous/LogReaderBase.cs, which was introduced in https://github.com/EDDiscovery/EDDiscovery/pull/294
This keeps track of the file position of the start of the line rather than the size of the file, and avoids reading the last line in the file if it doesn't end with a newline.

**Comment by richardbuckle** [comment]
Merged into Chapter4.
